cmake/Modules/FunctionQbtCXXCompilerAndModeCheck.cmake

```diff
@@ -0,0 +1,19 @@
+function(qbt_cxx_compiler_and_mode_check _min_std)
+    # ensure the compiler fully supports the minimum required C++ mode.
+    if(NOT CMAKE_CXX${_min_std}_STANDARD__HAS_FULL_SUPPORT)
```

Comment:
What happens if you set CMAKE_CXX_STANDARD to 14 without this check but compiler doesn't support it?

Reply:
@glassez The problem is that some compiler versions report that they "support" C++14, but it is not actually _full support_. So without this check, when using those compilers, the configure might complete successfully, but the build will fail later on.

Reply:
On another note, if we ever switch to c++17 we might not need full support c++17. For example, currently the c++17 full support check would probably fail for msvc2017, although we not use the extra features implemented in msvc2019. Am I wrong?
Of course, this worry is a bit theoretical at this timepoint.

Reply:
@sledgehammer999 
> although we not use the extra features

_currently_.

I don't think we should constrain ourselves because of outdated/broken toolchains. If we switch to C++17 but continue to support toolchains without full support for it, people making future contributions after that would have to be constantly walking on eggshells to make sure they don't use anything that happens to be unsupported by those toolchains, which sounds like a huge pain.

It won't be too harsh to require >=MSVC2019 to build by the time we switch to C++17 IMO, since that is still kind of in the distant future. By that time, Qt6 will also be released, and that is also set to require full C++17 support minimum AFAIK. So people will have time and reasons to update their stuff (I am assuming, of course, that we will eventually want to transition to Qt6).